src/api/hooks/useGetProductDetail.ts

```diff
@@ -0,0 +1,56 @@
+import type { UseQueryResult } from '@tanstack/react-query';
+import { useQuery } from '@tanstack/react-query';
+
+import type { GoodsData } from '@/types';
+
+import { fetchInstance } from '../instance';
+
+// Data type 정의
+export type ProductDetailData = GoodsData & {
+  isAccessableProductPage: boolean;
+  review: {
+    averageRating: number;
+    totalReviewCount: number;
+  };
+  productDescription: {
+    images: string[];
+  };
+  productDetailInfo: {
+    announcements: {
+      displayOrder: number;
+      name: string;
+      value: string;
+    }[];
+    terms: {
+      displayOrder: number;
+      title: string;
+      description: string;
+    }[];
+  };
+};
+
+// Request parameter type 정의
+type ProductDetailRequestParams = {
+  productId: number;
+};
+
+// API endpoint path 정의
+const getProductDetailPath = ({ productId }: ProductDetailRequestParams) => {
+  return `/v1/products/${productId}/detail`;
+};
+
+// API 함수 정의
+export const getProductDetail = async (
+  params: ProductDetailRequestParams,
+): Promise<ProductDetailData> => {
+  const response = await fetchInstance.get(getProductDetailPath(params));
```

Comment:
아래처럼만 사용해도 충분할거 같은데 함수를 별도로 정의하신 이유가 있나요?

```suggestion
  const response = await fetchInstance.get(`/v1/products/${params.productId}/detail`);
```

Reply:
템플릿 코드에서 path 함수를 별도로 정의하는 방식을 보고, 저도 이와 같은 방식을 적용해 보았습니다.